src/resolve/resolved-dns-stub.c

```diff
@@ -528,6 +603,73 @@ static int manager_dns_stub_tcp_fd(Manager *m) {
         return m->dns_stub_tcp_fd = TAKE_FD(fd);
 }
 
+static int manager_dns_stub_tcp_fd_extra(Manager *m, DNSStubListenerExtra *l) {
+        _cleanup_free_ char *pretty = NULL;
+        _cleanup_close_ int fd = -1;
+        int r;
+
+        if (l->fd >= 0)
+                return 0;
+
+        fd = socket(socket_address_family(&l->address), SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (fd < 0) {
+                r = -errno;
+                goto fail;
+        }
+
+        r = set_dns_stub_common_socket_options(fd);
+        if (r < 0)
+                goto fail;
+
+        r = setsockopt_int(fd, IPPROTO_IP, IP_TTL, true);
```

Comment:
you are executing ipv4 sockopts on something that is possibly an ipv6 socket. this needs some conditionalization on the family used